File: tobehe/104.maximum-depth-of-binary-tree.cpp
/*
 * @lc app=leetcode id=104 lang=cpp
 *
 * [104] Maximum Depth of Binary Tree
 */
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode* root) {
        int depth = 0;
        
        TreeNode *p = root;

        if(root == NULL)
            return 0;
        
        int left = maxDepth(p->left);
        int right = maxDepth(p->right);

        depth += left > right?left:right;
        depth++;
        
        return depth;

    }
};

File: AnYang/Tree/104. Maximum Depth of Binary Tree.py
# 2019-07-29 Mon

class Solution:
    def maxDepth(self, root):
        if root is None: 
            return 0 
        else: 
            left_D = self.maxDepth(root.left) 
            right_D = self.maxDepth(root.right) 
            return max(left_D, right_D) + 1 
